src/cpp/fastdds/xtypes/type_representation/TypeObjectUtils.cpp

```diff
@@ -0,0 +1,2952 @@
+{
+    common_collection_header_consistency(complete_collection_header.common());
+    if (complete_collection_header.detail().has_value())
+    {
+        complete_type_detail_consistency(complete_collection_header.detail().value());
+    }
+}
+
+void TypeObjectUtils::complete_sequence_type_consistency(
+        const CompleteSequenceType& complete_sequence_type)
+{
+    empty_flags_consistency(complete_sequence_type.collection_flag());
+    complete_collection_header_consistency(complete_sequence_type.header());
+    complete_collection_element_consistency(complete_sequence_type.element());
+}
+
+void TypeObjectUtils::common_array_header_consistency(
+        const CommonArrayHeader& common_array_header)
+{
+    array_bound_seq_consistency(common_array_header.bound_seq());
+}
+
+void TypeObjectUtils::complete_array_header_consistency(
+        const CompleteArrayHeader& complete_array_header)
+{
+    common_array_header_consistency(complete_array_header.common());
+    complete_type_detail_consistency(complete_array_header.detail());
+}
+
+void TypeObjectUtils::complete_array_type_consistency(
+        const CompleteArrayType& complete_array_type)
+{
+    empty_flags_consistency(complete_array_type.collection_flag());
+    complete_array_header_consistency(complete_array_type.header());
+    complete_collection_element_consistency(complete_array_type.element());
+}
+
+void TypeObjectUtils::complete_map_type_consistency(
+        const CompleteMapType& complete_map_type)
+{
+    empty_flags_consistency(complete_map_type.collection_flag());
+    complete_collection_header_consistency(complete_map_type.header());
+    map_key_type_identifier_consistency(complete_map_type.key().common().type());
+    complete_collection_element_consistency(complete_map_type.key());
+    complete_collection_element_consistency(complete_map_type.element());
+}
+
+void TypeObjectUtils::common_enumerated_literal_consistency(
+        const CommonEnumeratedLiteral& common_enumerated_literal)
+{
+    enumerated_literal_flag_consistency(common_enumerated_literal.flags());
+}
+
+void TypeObjectUtils::complete_enumerated_literal_consistency(
+        const CompleteEnumeratedLiteral& complete_enumerated_literal)
+{
+    common_enumerated_literal_consistency(complete_enumerated_literal.common());
+    complete_member_detail_consistency(complete_enumerated_literal.detail());
+    if (complete_enumerated_literal.detail().ann_builtin().has_value())
+    {
+        throw InvalidArgumentError("Only @default_literal and @value builtin annotations apply to enum literals");
+    }
+}
+
+void TypeObjectUtils::complete_enumerated_literal_seq_consistency(
+        const CompleteEnumeratedLiteralSeq& complete_enumerated_literal_seq)
+{
+    std::set<int32_t> values;
+    std::set<MemberName> member_names;
+    bool default_member = false;
+    for (size_t i = 0; i < complete_enumerated_literal_seq.size(); i++)
+    {
+        if (complete_enumerated_literal_seq.size() == 0)
```

Comment:
The check for empty complete_enumerated_literal_seq should be done once at the beginning, not in every iteration.

Reply:
Done in 541798ae0